rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/HateoasResponseFilter.java

```diff
@@ -71,6 +72,11 @@
 @Filter("${restapi.hateoas.filter.uri}")
 @Requires(property = "restapi.hateoas.filter.enabled")
 public class HateoasResponseFilter implements HttpServerFilter {
+  private final List<LinkProvider> linkProviderList;
+
+  public HateoasResponseFilter(List<LinkProvider> linkProviderList) {
```

Comment:
What happens if a project does not provide any `LinkProvider`? The startup of that application should not fail. Meaning the `linkProviderList` should be consider as an optional dependency of `HateoasResponseFilter`

Reply:
Micronaut should inject an empty list if there are no beans of type LinkProvider